torch/fx/experimental/proxy_tensor.py

```diff
@@ -812,12 +812,7 @@ def can_handle_tensor(x: Tensor) -> bool:
 
     if func is torch.ops.aten.is_nonzero.default:
         with proxy_mode:
-            from .symbolic_shapes import guard_size_oblivious
-
-            if guard_size_oblivious(args[0].numel() != 1):  # type: ignore[attr-defined]
-                raise RuntimeError(
-                    "Boolean value of Tensor with more than one value is ambiguous"
-                )
+            torch._check(args[0].numel() == 1, lambda: "Boolean value of Tensor with more than one value is ambiguous")  # type: ignore[attr-defined]
```

Comment:
What does torch._check() do with unbacked? Does it just treat it as false?

Reply:
adds a runtime assert that's checked at runtime

Reply:
So is that the desired behavior? It seems like previously unbacked would always raise the runtime error (because of 0/1 specialization?).